src/limits/ThreadsLimitListener.cc

```diff
@@ -33,10 +35,19 @@ ThreadsLimitListener::ThreadsLimitListener(int32_t threadsLimit)
                 "clone",
                 seccomp::action::ActionAllow{},
                 (Arg(2) & CLONE_VM) == CLONE_VM));
+        syscallRules_.emplace_back(seccomp::SeccompRule(
+                "clone3",
+                seccomp::action::ActionAllow{},
+                (Arg(2) & CLONE_VM) == CLONE_VM));
```

Comment:
I added `ENOSYS` for `clone3`, but now this test is failing:
```
1: ======================================================================
1: ERROR: test_1_sec_program_threads_1 (testsuits.test_times.TestReportedTimes)
1: ----------------------------------------------------------------------
1: Traceback (most recent call last):
1:   File "/sio2jail/test/testsuits/test_times.py", line 24, in test_1_sec_program_threads_1
1:     self.assertAlmostEqual(result.time, 1.0)
1:   File "/usr/local/lib/python3.9/unittest/case.py", line 868, in assertAlmostEqual
1:     diff = abs(first - second)
1: TypeError: unsupported operand type(s) for -: 'NoneType' and 'float'
1: -------------------- >> begin captured stdout << ---------------------
1: running:
1: /sio2jail/build/./src/sio2jail -b /sio2jail/build/./boxes/./minimal/:/:ro -t 1 -m 1G /sio2jail/build/./test/src/1-sec-prog-th flat 1
1:
1: result: 2
1:
1: stdout:
1:
1: stderr:
1: Exception occurred: System error occured: ptrace setopts child failed: No such process: error 3: No such process
1:
1:
1: --------------------- >> end captured stdout << ----------------------
1:
1: ----------------------------------------------------------------------
1: Ran 14 tests in 1.575s
1:
1: FAILED (errors=1)
1/1 Test #1: python ...........................***Failed    1.75 sec
```

When `clone3` is allowed, this test passes

Reply:
ok, I think I know why this happens.

TraceExecutor gets clone events here
https://github.com/sio2project/sio2jail/blob/8e65e31/src/tracer/TraceExecutor.cc#L94-L95

not due to seccomp traps, but due to `PTRACE_O_TRACECLONE`.

It then tries to get child's pid, and set up ptrace of the child too:
https://github.com/sio2project/sio2jail/blob/8e65e3114acc8ba3d3daff3635c46ad933fa9a6c/src/tracer/TraceExecutor.cc#L137-L156

I suspect this triggers even if the a clone (in this case `clone3`) fails, and TraceExecutor doesn't check for that.

Maybe ptrace has some way to get the clone's error code, so that we can check if it succeeded.
If there's no way to do that, I guess we can just catch the `ESRCH` returned by `PTRACE_SETOPTIONS`, assume clone failed and move on, so that the process can retry with the old `clone`.